src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js

```diff
@@ -0,0 +1,164 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import _ from 'lodash/fp'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import { azureMachineTypes, defaultAzureMachineType } from 'src/libs/azure-utils'
+import { formatUSD } from 'src/libs/utils'
+import {
+  defaultAzureWorkspace, imageDocs, testAzureDefaultRegion
+} from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import { getAzureComputeCostEstimate, getAzureDiskCostEstimate } from 'src/pages/workspaces/workspace/analysis/cost-utils'
+import { toolLabels } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { asMockedFn } from 'src/testing/test-utils'
+
+import { AzureComputeModalBase } from './AzureComputeModal'
+
+
+jest.mock('src/pages/workspaces/workspace/analysis/cost-utils')
+
+jest.mock('src/libs/ajax')
+jest.mock('src/libs/notifications', () => ({
+  notify: jest.fn()
+}))
+const onSuccess = jest.fn()
+const defaultModalProps = {
+  onSuccess, onDismiss: jest.fn(), onError: jest.fn(),
+  currentRuntime: undefined, currentDisk: undefined, tool: toolLabels.JupyterLab, workspace: defaultAzureWorkspace,
+  location: testAzureDefaultRegion
+}
+
+
+const defaultAjaxImpl = {
+  Runtimes: {
+    runtime: () => ({
+      details: jest.fn()
+    })
+  },
+  Buckets: { getObjectPreview: () => Promise.resolve({ json: () => Promise.resolve(imageDocs) }) },
+  Disks: {
+    disk: () => ({
+      details: jest.fn()
+    })
+  },
+  Metrics: {
+    captureEvent: () => jest.fn()
+  }
+}
+
+const verifyEnabled = item => expect(item).not.toHaveAttribute('disabled')
+
+describe('AzureComputeModal', () => {
+  beforeAll(() => {
+  })
+
+  beforeEach(() => {
+    // Arrange
+    Ajax.mockImplementation(() => ({
+      ...defaultAjaxImpl
+    }))
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const getCreateButton = () => screen.getByText('Create')
+  it('renders correctly with minimal state', async () => {
+    // Arrange
+
+    // Act
+    await act(async () => await render(h(AzureComputeModalBase, defaultModalProps)))
```

Comment:
I see a lot of use of the act() wrapper in this PR.... Are they really necessary?  Overuse of act is a common mistake.  Try removing these and seeing if the tests work fine without them.

Reply:
Removing any of them prompts a warning from Jest that React state changes should be wrapped in an act(); it seems that even a plain render operation causes a `setBusy()` call which updates state. In fact there was one place where render was being awaited without an act where I had to *add* an `act()` wrapper to remove the warning. I'm of course open to alternatives but it seems like all of them are needed?

Reply:
no worries...I just don't want folks seeing this code to think an //Act section always needs an act() wrapper... an easy false correlation.   Please add a comment to these to say something like "wrapping component init-time side-effects with act"

Reply:
Wonderful, I will do so! Thanks for clarifying; I'm still learning React+Jest so this is helpful for me